js/modules/k6/http/http_request.go

```diff
@@ -389,9 +389,11 @@ func (h *HTTP) request(ctx context.Context, preq *parsedHTTPRequest) (*HTTPRespo
 		}
 	}
 
+	transport := netext.NewTransport(state.Transport, state.Samples, state.Options, tags)
```

Comment:
Why not just pass the `state` (a single pointer) instead of 3+ of its components (`Options` especially being a beefy struct that would be copies on every request)? And it would probably be a bit more readable, especially if, as I mentioned in the other comments, the `ntlmAuthCache` is moved in the `state`?